cpp/src/arrow/array-binary-test.cc

```diff
@@ -359,300 +232,161 @@ class TestBinaryArray : public ::testing::Test {
-    if (valid_bytes_[i] == 0) {
-      ASSERT_TRUE(strings_->IsNull(i));
-    } else {
-      ASSERT_FALSE(strings_->IsNull(i));
-      ASSERT_EQ(strings_->GetString(i), expected_[i]);
-    }
-  }
-  TestInitialized(*strings_);
+TYPED_TEST(TestStringArray, TestEmptyStringComparison) {
+  this->_TestEmptyStringComparison();
 }
 
-TEST_F(TestBinaryArray, TestPaddingZeroed) { AssertZeroPadded(*strings_); }
+TYPED_TEST(TestStringArray, CompareNullByteSlots) { this->_TestCompareNullByteSlots(); }
 
-TEST_F(TestBinaryArray, TestGetString) {
-  for (size_t i = 0; i < expected_.size(); ++i) {
-    if (valid_bytes_[i] == 0) {
-      ASSERT_TRUE(strings_->IsNull(i));
-    } else {
-      std::string val = strings_->GetString(i);
-      ASSERT_EQ(0, std::memcmp(expected_[i].data(), val.c_str(), val.size()));
-    }
-  }
-}
+TYPED_TEST(TestStringArray, TestSliceGetString) { this->_TestSliceGetString(); }
 
-TEST_F(TestBinaryArray, TestEqualsEmptyStrings) {
-  BinaryBuilder builder;
-
-  std::string empty_string("");
-  for (int i = 0; i < 5; ++i) {
-    ASSERT_OK(builder.Append(empty_string));
-  }
-
-  std::shared_ptr<Array> left_arr;
-  FinishAndCheckPadding(&builder, &left_arr);
-
-  const BinaryArray& left = checked_cast<const BinaryArray&>(*left_arr);
-  std::shared_ptr<Array> right =
-      std::make_shared<BinaryArray>(left.length(), left.value_offsets(), nullptr,
-                                    left.null_bitmap(), left.null_count());
-
-  ASSERT_TRUE(left.Equals(right));
-  ASSERT_TRUE(left.RangeEquals(0, left.length(), 0, right));
-}
+// ----------------------------------------------------------------------
+// String builder tests
 
-class TestBinaryBuilder : public TestBuilder {
+template <typename T>
+class TestStringBuilder : public TestBuilder {
  public:
+  using TypeClass = T;
+  using offset_type = typename TypeClass::offset_type;
+  using ArrayType = typename TypeTraits<TypeClass>::ArrayType;
+  using BuilderType = typename TypeTraits<TypeClass>::BuilderType;
+
   void SetUp() {
     TestBuilder::SetUp();
-    builder_.reset(new BinaryBuilder(pool_));
+    builder_.reset(new BuilderType(pool_));
   }
 
   void Done() {
     std::shared_ptr<Array> out;
     FinishAndCheckPadding(builder_.get(), &out);
 
-    result_ = std::dynamic_pointer_cast<BinaryArray>(out);
+    result_ = std::dynamic_pointer_cast<ArrayType>(out);
     ASSERT_OK(ValidateArray(*result_));
   }
 
- protected:
-  std::unique_ptr<BinaryBuilder> builder_;
-  std::shared_ptr<BinaryArray> result_;
-};
-
-TEST_F(TestBinaryBuilder, TestScalarAppend) {
-  std::vector<std::string> strings = {"", "bb", "a", "", "ccc"};
-  std::vector<uint8_t> is_valid = {1, 1, 1, 0, 1};
+  void _TestScalarAppend() {
+    std::vector<std::string> strings = {"", "bb", "a", "", "ccc"};
+    std::vector<uint8_t> is_valid = {1, 1, 1, 0, 1};
 
-  int N = static_cast<int>(strings.size());
-  int reps = 10;
+    int N = static_cast<int>(strings.size());
+    int reps = 1000;
```

Comment:
any reason why reps increased?

Reply:
Hmm... Apparently various values of `reps` were originally used. I'll change back to them.